compute_endpoint/globus_compute_endpoint/endpoint/interchange.py

```diff
@@ -360,11 +366,33 @@ def process_pending_tasks():
                     if self.time_to_quit:
                         self.stop()
                         continue  # nominally == break; but let event do it
+
                     try:
-                        incoming_task = self.pending_task_queue.get(timeout=1)
-                        task_msg = unpack(incoming_task)
-                        if not isinstance(task_msg, Task):
-                            raise InvalidMessageError()
+                        prop_headers, body = self.pending_task_queue.get(timeout=1)
+
```

Comment:
@yadudoc Kevin called to attention that we were trying to avoid unpacking the message, and should do so in the worker instead.  Removed this unpack as we can get the task_uuid from the headers.  Just wanted to call it to your attention.  (Also see below in process_pending_results)

Reply:
Neat, @khk-globus and I had talked and confirmed that this is indeed what we'd like to do.

Reply:
Looks good.  Thanks for double-checking this, @yadudoc !